### PR TITLE
Feature/webshop 7 refactor and create other validators 2

### PR DESCRIPTION
**Pull request by deb05121**

### CODE REVIEW

File: webshop-backend/src/shopping/ShoppingCart.java [hunk 1]
```diff
@@ -23,8 +23,11 @@ public User getOwner() {
...
         boolean ok = false;
         if (p != null) {
-            products.add(p);
-            p.decreaseStock();
+            //nem lehet 0 a stock, mert abból nem tudok levonni darabszámot
+            if(p.getStock()!=0){
+                products.add(p);
+                p.decreaseStock();
+            }
             ok = true;
```
Comment (koics97) [inline webshop-backend/src/shopping/ShoppingCart.java]: why is `ok = true;` outside the `if` statement ?
I'm sure you don't want to return `true` if the `if(p.getStock()!=0)` statement is false.
Or am I wrong ?

File: webshop-backend/src/shopping/ShoppingCart.java [hunk 2]
```diff
@@ -35,8 +38,12 @@ public boolean removeProduct(Product p) {
...
             if (pieces > 0) {
-                products.remove(p);
-                p.increaseStock();
+
+                //nem lehet -1 a kosár tartalma
+                if(products.contains(p)){
+                    products.remove(p);
+                    p.increaseStock();
+                }
                 ok = true;
```
Comment (koics97) [inline webshop-backend/src/shopping/ShoppingCart.java:47]: Same thing here

Comment (koics97) [inline webshop-backend/src/shopping/ShoppingCart.java]: Same thing here as in `addProduct()`

File: webshop-backend/src/shopping/ValidatorTest.java [hunk 3]
```diff
@@ -1,35 +1,63 @@
...
+    void isCityValidatorTrue(String city) {
+        Assertions.assertTrue(cityValidator.isValid(city));
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"bp", "debrecen", "budaPest"})
+    void isCityValidatorFalse(String city) {
+        Assertions.assertFalse(cityValidator.isValid(city));
+    }
 }
```
Comment (koics97) [inline webshop-backend/src/shopping/ValidatorTest.java:63]: I like your tests. Like a real pro :)

File: webshop-backend/src/shopping/ShoppingCart.java (hunk 1 above)
Comment (deb05121) [reply webshop-backend/src/shopping/ShoppingCart.java]: Right. I will update shoppingCart if it is allowed in the validator branch?! 
Did you read my review?
I would like to write a setProductListEmpty(){products.clear()} method.

File: webshop-backend/src/shopping/ShoppingCart.java (hunk 2 above)
Comment (deb05121) [reply webshop-backend/src/shopping/ShoppingCart.java]: ok

File: webshop-backend/src/shopping/ValidatorTest.java (hunk 3 above)
Comment (deb05121) [reply webshop-backend/src/shopping/ValidatorTest.java:63]: thanks

File: webshop-backend/src/shopping/ShoppingCart.java (hunk 1 above)
Comment (lyancsie) [reply webshop-backend/src/shopping/ShoppingCart.java]: @deb05121, sure, you can fix these one-liners here. If you notice something that requires more job, it is better to open a new branch, but in the case of these 1-line changes, I think it is okay :)

**Comment by deb05121** [comment]
Hey guys! There is a big problem.
I don't see anyone instantiating MonetaryAmount.
The price cannot stand alone. That isn't just an interger. This is a MonetaryAmount instance.
There will be a MonetaryAmount instance in everyone product instance.
The same would be in the User class.

**Comment by deb05121** [comment]
Hi @koics97 @KJudit01 @hollszabolcs 
I will merge if you agree.

**Review by koics97 — APPROVED** [review]
Looks clear to me

**Pull request merged**